src/bci_build/package/nginx/README.md.j2

```diff
@@ -0,0 +1,103 @@
+
+```ShellSession
+$ podman run -it --rm -p 8080:80 {{ image.pretty_reference }}
+```
+
+Or:
+
+```ShellSession
+$ podman run -it --rm -p 8080:80 -v /path/to/html/:/srv/www/htdocs/:Z {{ image.pretty_reference }}
+```
+
+**Note:** The directory `/srv/www/htdocs/` is the root directory used by the default server. Additional servers can use any other path.
+
+Once the container is running, access `http://localhost:8080` or `http://host-ip:8080` in your browser.
+
+## Environment variables
+
+### NGINX_ENTRYPOINT_QUIET_LOGS
+
+This optional environment variable controls the logging during container startup. Set the value to `1` to silence logs.
+
+### NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE
+
+This optional environment variable enables a script to autotune the number of worker processes. Set the value to `1` to enable autotune of the worker process parameter.
+
+## Using templates
+
+Out of the box, nginx doesn't support environment variables inside configuration blocks, but this image has a script that can extract environment variables before nginx starts.
+
+By default, this script reads template files in `/etc/nginx/templates/*.template` and outputs the result of executing `envsubst` to `/etc/nginx/conf.d`.
+
+If you create a file named `/etc/nginx/templates/default.conf.template`, which contains a variable reference like this:
+
+```
+listen       ${NGINX_PORT};
+```
+
+The script will output it to `/etc/nginx/conf.d/default.conf` like this:
+
+```
+listen       80;
+```
+
+The use of templates allows you to set arbitrary environment variables that can be replaced inside the container, such as `-e NGINX_PORT=80`.
+
+### NGINX_ENVSUBST_TEMPLATE_DIR
+
+This optional environment variable changes the directory that contains template files (default: `/etc/nginx/templates`).
+
+**Note:** The script will ignore template processing if this directory doesn't exist
+
+### NGINX_ENVSUBST_OUTPUT_DIR
+
+This optional environment variable changes the directory where the result of executing `envsubst` on templates is output (default: `/etc/nginx/conf.d`).
+
+The output filename is the template filename with the suffix removed.
+
+### NGINX_ENVSUBST_TEMPLATE_SUFFIX
+
+This optional environment variable changes the suffix of template files (default: `.template`).
+
+**Note:** The script only processes files whose name ends with this suffix.
+
+### NGINX_ENVSUBST_STREAM_OUTPUT_DIR
+
+This optional environment variable changes the directory where the result of executing `envsubst` on stream templates is output (default: `/etc/nginx/stream-conf.d`).
+
+The output filename is the template filename with the suffix removed.
+
+### NGINX_ENVSUBST_STREAM_TEMPLATE_SUFFIX
+
+This optional environment variable changes the suffix of stream template files (default: `.stream-template`).
+
+**Note:** The script only processes files whose name ends with this suffix.
```

Comment:
What are these stream templates? Something like the "ordinary" templates but with a different destination?

Reply:
Yes, same thing, just another path, but now I wonder, our default `nginx.conf` does not even load this stream path, just `vhosts.d/*.conf` and `conf.d/*.conf`.

Maybe not worth documenting this, unless we explicitly mention that this will only work if you change `nginx.conf`.

Reply:
I removed this part, not worth having as our default config does not support it.